## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [14]:
import pandas as pd
import numpy as np
df = pd.read_csv('data/data.csv')
data = np.array(df['data'])
data

array([17.49945292, 19.66239904,  7.18282343, 29.84162485,  9.23938649,
       14.94951581, 23.0172885 , 12.52827148, 13.65228669, 14.54595431,
       23.30872427, 33.75324808, 20.24923636, 13.29244733, 23.23434992,
       16.4230418 , 19.88521702, 27.05000732, 15.5127743 , 20.05415151,
       14.73135264, 19.06139498, 21.53942271, 14.06732571, 17.9670682 ,
       18.58289581, 16.17406993, 12.87432628, 11.47269664, 19.07902883,
       18.38565824, 33.38820073,  5.39139454, 20.67635903, 22.22266722,
       28.15780318, 23.01114324, 14.93471778, 20.00005857, 23.25411543,
       18.11895082, 24.62607043,  8.79145607, 30.387108  , 28.80606806,
       17.98593597, 23.66804468, 20.28782355, 15.02518827, 20.52626131,
       26.00219532, 17.71344489, 17.74598346, 19.55317542, 22.60097798,
       27.67027538, 16.19192417, 23.05037746, 21.29669604,  8.84832568,
       17.48410111, 19.20602661, 19.76257856, 21.9560206 ,  7.75806171,
       20.27753314, 15.93394654, 11.36336584, 23.14577858, 24.41

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [16]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [22]:
from statsmodels.stats.weightstats import ztest
#z-test бо даних більше 30 
ztest_Score, p_value = ztest(data, value = 20, alternative='larger')
ztest_Score, p_value

(-0.9953477389335053, 0.8402164165464012)

Використано код для одновибіркового z-тесту, де ми перевіряємо гіпотезу:

Нульова гіпотеза H0 середній чек дорівнює 20 доларів
Альтернативна гіпотеза На середній чек статистично значущо більший за 20 доларів

Z-статистика: -0.9953477389335053
Це значення z-тесту показує відхилення середнього значення зразка від гіпотетичного значення $20 в одиницях стандартних відхилень.

p-значення: 0.8402164165464012
Це значення показує ймовірність того, що ми спостерігаємо отриманий або ще більш екстремальний результат, якщо нульова гіпотеза правдива.
Оскільки p-значення 0.84 значно більше за звичайний поріг (наприклад, 0.05 або 0.01), ми не можемо відкинути нульову гіпотезу.

Висновок:
На основі результатів z-тесту, з p-значенням 0.84, ми не маємо підстав стверджувати, що середній чек за липень 2022 року статистично значущо перевищує $20. Тобто, статистичних доказів підвищення середнього чеку немає.

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [32]:
df_2 = pd.read_csv('data/data_2.csv')

In [46]:
ztest_score, p_value = ztest(df, df_2, alternative='two-sided')

print("Середній чек липня:", df.mean())
print("Середній чек серпня:", df_2.mean())
print(f"Z-тест: {ztest_score}, p-значення: {p_value}")

Середній чек липня: Unnamed: 0    49.500000
data          19.377553
dtype: float64
Середній чек серпня: data    19.373223
dtype: float64
Z-тест: [1.11025455e+01 5.24667346e-03], p-значення: [1.21921153e-28 9.95813779e-01]




Висновок:

За p-значенням 0.9958, ми не маємо підстав відкидати нульову гіпотезу. Тобто, немає статистично значущої різниці між середніми чеками за липень і серпень 2022 року.

Збіг висновків: Середні значення за липень і серпень дуже близькі (19.38 проти 19.37 доларів). Це узгоджується з результатами статистичного тесту — немає суттєвої різниці між середніми значеннями.

Отже, згідно з результатами, чек у серпні практично не змінився порівняно з липнем, і різниця не є статистично значущою